base/sim/FairMCApplication.cxx

```diff
@@ -1007,9 +1007,8 @@ void FairMCApplication::ConstructGeometry()
     delete particleIter;
   }
   fModIter->Reset();
-  while((Mod = dynamic_cast<FairModule*>(fModIter->Next()))) {
-    Mod->ModifyGeometry();
-  }
+
+  fRun->AlignGeometry();
```

Comment:
1. Have you considered calling `fRun->AlignGeometry()` in the [TVirtualMCApplication::MisalignGeometry() hook](https://root.cern.ch/doc/master/classTVirtualMCApplication.html#a5f0c903039eb1e4407e7a9a22dbddcad)?
2. Not calling our `ModifyGeometry()` hook any more will break some examples, won't it? If the new alignment module proposed here is intended to replace the `FairModule::ModifyGeometry()` hook, then there is more work needed, I am afraid (deprecate the interface and update the examples). If you do not want to update the examples, then we should still call it to not break existing code (probably it should also be moved to the `TVirtualMCApplication::MisalignGeometry()` hook!?)

Reply:
1. No did not know that existed. It looks like that would be the right place to call it! Why have you not used that before and called `Mod->ModifyGeometry();` from within `ConstructGeometry()`?
2. I think it should replace the `FairModule::ModifyGeometry()` hook, since I do not see the need of ModifyGeometry() anymore. Is that ok for you? Then we would deprecate it and update the examples (and yes move the `fRun->AlignGeometry();` call to the `TVirtualMCApplication::MisalignGeometry()` hook).

Reply:
I just checked how you handle the deprecation. As far as I saw you just use a `LOG(WARNING)` message at runtime, is that correct?

Reply:
for deprecation warning, use `__attribute__((deprecated("message")`, which will warn at compile time if the marked method is used.
Example:
`void ModifyGeometry() __attribute__((deprecated("Use AlignGeometry instead")))`;

Reply:
Ok, that looks better. Btw, does FairRoot require any c++ standard, I believe c++11 or?

Reply:
> Ok, that looks better. Btw, does FairRoot require any c++ standard, I believe c++11 or?

Correct.

Reply:
>     1. No did not know that existed. It looks like that would be the right place to call it! Why have you not used that before and called `Mod->ModifyGeometry();` from within `ConstructGeometry()`?

Probably because I overlooked the existence of this hook. 
 
>     2. I think it should replace the `FairModule::ModifyGeometry()` hook, since I do not see the need of ModifyGeometry() anymore. Is that ok for you? Then we would deprecate it and update the examples (and yes move the `fRun->AlignGeometry();` call to the `TVirtualMCApplication::MisalignGeometry()` hook).

Here I completely agree. To my knowledge the ModifyGeometry is only used in the FairRoot examples.

Reply:
My current workaround for this is indeed to run the macro twice.

1. Call the Init() function, get misalignment parameters and store them to disk (ascii file etc.) but don't run the actual simulation
2. Read the parameters from disk before calling Init(), pass them to the FairAlignHandler and then call Init() to apply the misalignment.

I updated the Tutorial4 to reflect that and removed all the code that is no longer neccessary.